### ISSUE TITLE
Single Pixel Attack not working

### ISSUE THREAD

**Issue by hagianga21** [bethgelab/foolbox#341]
Hi, I am testing with Single Pixel Attack. I run the code in Colab. My code is simple:

!pip install foolbox
from keras.models import load_model
from keras.applications.vgg16 import VGG16
import foolbox
from foolbox.models import KerasModel
from foolbox.attacks import SinglePixelAttack
import numpy as np
import foolbox
from foolbox.criteria import Misclassification, TargetClass
import keras
from keras.applications.vgg16 import preprocess_input

model = VGG16(weights='imagenet')
image, label = foolbox.utils.imagenet_example()
mean = [103.939, 116.779, 123.68]  # BGR
std = 1
fmodel = foolbox.models.KerasModel(model, bounds=(0, 255), preprocessing=(mean, std))
attack = foolbox.attacks.SinglePixelAttack(model=fmodel, criterion=Misclassification)
adversarial = attack(image[..., ::-1], label = label, unpack=False, max_pixels = 1)

The results showed error:
is_adversarial() missing 1 required positional argument: 'label'

**Comment by FlashTek** [comment]
Hi @hagianga21, this is not a issue of `foolbox` but rather of how you call the attack function:
when you create the attack you must pass an instance of the criterion and not the class, i.e. you need to change `criterion=Misclassification` to `criterion=Misclassification()`. 

```
!pip install foolbox
from keras.models import load_model
from keras.applications.vgg16 import VGG16
import foolbox
from foolbox.models import KerasModel
from foolbox.attacks import SinglePixelAttack
import numpy as np
import foolbox
from foolbox.criteria import Misclassification, TargetClass
import keras
from keras.applications.vgg16 import preprocess_input

model = VGG16(weights='imagenet')
image, label = foolbox.utils.imagenet_example()
mean = [103.939, 116.779, 123.68] # BGR
std = 1
fmodel = foolbox.models.KerasModel(model, bounds=(0, 255), preprocessing=(mean, std))
attack = foolbox.attacks.SinglePixelAttack(model=fmodel, criterion=Misclassification())
adversarial = attack(image[..., ::-1], label = label, unpack=False, max_pixels = 1)
```

**Comment by jonasrauber** [comment]
@FlashTek I agree, we should add an assertion, but to follow pythons duck typing philosophy, we should probably just check whether the object has an `is_adversarial` method rather than enforcing the type (we can however perform an additional check to explicitly catch uninitialized sub-classes of the `Criterion` class and give an even better error message in that case). PR welcome!

**Comment by jonasrauber** [comment]
Created a new issue for the additional checks we should add.
